src/test/claimtriedb_tests.cpp

```diff
@@ -0,0 +1,127 @@
+#include "claimtrie.h"
+#include "main.h"
+#include "uint256.h"
+
+#include "test/test_bitcoin.h"
+#include <boost/test/unit_test.hpp>
+using namespace std;
+
+#include "claimtriedb.cpp" /// avoid linker problems
+
+struct claimtriedb_test_helper
+{
+    static void migrate(CClaimTrieDb &db)
+    {
+        db.doMigrate();
+    }
+};
+
+BOOST_FIXTURE_TEST_SUITE(claimtriedb_tests, BasicTestingSetup)
+
+
+BOOST_AUTO_TEST_CASE(hash_test)
+{
+    const size_t intclaimQueueRowType = hashType<int, claimQueueRowType>();
+    const size_t intsupportQueueRowType = hashType<int, supportQueueRowType>();
+    const size_t stringCClaimTrieNode = hashType<std::string, CClaimTrieNode>();
+    const size_t stringqueueNameRowType = hashType<std::string, queueNameRowType>();
+    const size_t intexpirationQueueRowType = hashType<int, expirationQueueRowType>();
+    const size_t uint160CClaimIndexElement = hashType<uint160, CClaimIndexElement>();
+    const size_t stringsupportMapEntryType = hashType<std::string, supportMapEntryType>();
+    const size_t stringsupportQueueNameRowType = hashType<std::string, supportQueueNameRowType>();
+    const size_t intsupportExpirationQueueRowType = hashType<int, supportExpirationQueueRowType>();
+
+    BOOST_CHECK(intclaimQueueRowType                == 2291382435);
+    BOOST_CHECK(stringCClaimTrieNode                == 2689964764);
+    BOOST_CHECK(intsupportQueueRowType              == 2519094245);
+    BOOST_CHECK(stringqueueNameRowType              == 2566197209);
+    BOOST_CHECK(stringsupportMapEntryType           == 212036296);
+    BOOST_CHECK(intexpirationQueueRowType           == 2977646682);
+    BOOST_CHECK(uint160CClaimIndexElement           == 1885673905);
+    BOOST_CHECK(stringsupportQueueNameRowType       == 389061475);
+    BOOST_CHECK(intsupportExpirationQueueRowType    == 3108630299);
+}
```

Comment:
I really wish we could avoid this kind of test, but if we go this route, at least it helps locate regressions.

Reply:
The test aims to *guarantee* that all distinct type has an unique id, when new type is added unit test should be updated as well. The algorithm should produce same results on all different compilers and all its version, which is *not* guaranteed nor by std::type_id nor boost::type_id.

Reply:
This is not a strong argument for why we want/need to support our own hashing here. This seems like a case of NIH, which should be avoided (https://en.wikipedia.org/wiki/Not_invented_here#In_computing).  I don't have an understanding of why we *must* go this route without considering other options.  What should be an organizational PR (as the name states separating methods out), looks to be far larger, including a breaking database re-organization which is something else entirely.

@bvbfan If you've had discussions about this work that I was not a part of (surely you have), I'd like to chat with whoever had them with you :-)  @kaykurokawa ?  If it's just a matter of me not being caught-up, I'll adjust the review accordingly.